### PR TITLE
feat: add e-mail as a required field

### PR DESCRIPTION
**Pull request by sutt0n**
Updates:
- feat: added `email` as a required and validity-enforced field for a marketplace post
- feat: dded translations for `email` and its error states
- fix: keyboard hiding description box entirely, making the user click away to unfocus and hide the keyboard to continue
- fix: spacing on first and last screens between the fields

Dev note: Most of the bulk changes you see in a lot of the files are from `prettier`.

image
image
image

### CODE REVIEW

File: app/modules/market-place/screens/create-post/create-post-screen.tsx [hunk 1]
```diff
@@ -12,191 +12,226 @@ import {
...
+
+    if (!email) {
+      setEmailError(t.marketPlace.email_is_required());
+    } else if (
+      /^((?!\.)[\w\-_.]*[^.])(@\w+)(\.\w+(\.\w+)?[^.\W])$/.test(email) === false
+    ) {
+      setEmailError(t.marketPlace.email_is_invalid());
+    } else {
+      emailValid = true;
+      setEmailError("");
```
Comment (sutt0n) [inline app/modules/market-place/screens/create-post/create-post-screen.tsx:82]: I added this.

File: app/modules/market-place/screens/create-post/create-post-screen.tsx [hunk 2]
```diff
@@ -12,191 +12,226 @@ import {
...
 
-  const [nameError, setNameError] = useState("")
-  const [descriptionError, setDescriptionError] = useState("")
-  const [filteredTags, setFilteredTags] = useState<MarketplaceTag[]>([])
-  const [selectedTags, setSelectedTags] = useState<MarketplaceTag[]>([])
-  const [initTag, setInitTag] = useState<MarketplaceTag[]>([])
-  const timeoutRef = useRef(null)
-  const { LL: t } = useI18nContext()
+  const [nameError, setNameError] = useState("");
+  const [emailError, setEmailError] = useState("");
```
Comment (sutt0n) [inline app/modules/market-place/screens/create-post/create-post-screen.tsx:52]: This was also added.

File: app/modules/market-place/screens/create-post/create-post-screen.tsx [hunk 3]
```diff
@@ -12,191 +12,226 @@ import {
...
-  const dispatch = useDispatch()
-  const [name, setName] = useState("")
-  const [description, setDescription] = useState("")
-  const [isLoading, setIsLoading] = useState(false)
-  const [tagLoading, setTagLoading] = useState(false)
-  const tempPost = useSelector((state: RootState) => state.storeReducer.tempPost)
-  const [tag, setTag] = useState("")
+  const dispatch = useDispatch();
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
```
Comment (sutt0n) [inline app/modules/market-place/screens/create-post/create-post-screen.tsx:42]: This was added.

File: app/modules/market-place/screens/create-post/create-post-screen.tsx [hunk 4]
```diff
@@ -216,9 +251,25 @@ export const CreatePostScreen: React.FC<Props> = ({ navigation }) => {
...
+              <TextInputComponent
+                onChangeText={setEmail}
+                value={email}
+                placeholder={t.marketPlace.email()}
+                isError={emailError !== ""}
+              />
+
+              {emailError
+                ? <Text style={styles.errorText}>{emailError}</Text>
+                : null}
```
Comment (sutt0n) [inline app/modules/market-place/screens/create-post/create-post-screen.tsx:268]: This was added.

File: app/modules/market-place/screens/find-location-screen/find-location-screen.tsx [hunk 5]
```diff
@@ -160,11 +169,10 @@ export const FindLocationScreen: React.FC<Props> = ({ navigation }) => {
                 dispatch(
                   setTempPost({
                     ...tempPost,
-                    email: "TestMail@gmail.com",
```
Comment (sutt0n) [inline app/modules/market-place/screens/find-location-screen/find-location-screen.tsx:163]: Valid delta.

**Review by miladezzat — APPROVED** [review]

File: app/modules/market-place/screens/post-detail-screen/PostDetailScreen.tsx [hunk 6]
```diff
@@ -73,13 +81,19 @@ const DetailComponent = ({ editable, isHidePhone, setIsHidePhone, post }: Detail
       <Text style={detailStyle.label}>{t.marketPlace.description()}</Text>
       <Text style={detailStyle.value}>{post?.description}</Text>
 
+      <Text style={detailStyle.label}>{t.marketPlace.email()}</Text>
+      <Text style={detailStyle.value}>{post?.email}</Text>
+
```
Comment (sutt0n) [inline app/modules/market-place/screens/post-detail-screen/PostDetailScreen.tsx:86]: Valid delta.

**Pull request merged**